File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans = 0
        lst = []
        for i in range(len(s)):
            if s[i] not in lst:
                lst.append(s[i])
                if i == len(s)-1:
                    return max(ans, len(lst))                
            else:
                ans = max(ans, len(lst))
                ind = lst.index(s[i])
                lst = lst[ind+1:]
                lst.append(s[i])
        return ans
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        List<Character> seq = new ArrayList<>();
        int len = 0;
        for(int i = 0; i < s.length(); i++) {
            if(seq.contains(s.charAt(i))) {
                len = Math.max(len, seq.size());
                int idx = seq.indexOf(s.charAt(i));
                seq = seq.subList(idx+1, seq.size());
            }
            seq.add(s.charAt(i));
            // System.out.println(seq);
        }
        return Math.max(len, seq.size());
    }
}